pkg/phlaredb/profile_store.go

```diff
@@ -0,0 +1,452 @@
+package phlaredb
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/runutil"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"github.com/segmentio/parquet-go"
+	"go.uber.org/atomic"
+
+	phlaremodel "github.com/grafana/phlare/pkg/model"
+	phlarecontext "github.com/grafana/phlare/pkg/phlare/context"
+	"github.com/grafana/phlare/pkg/phlaredb/block"
+	schemav1 "github.com/grafana/phlare/pkg/phlaredb/schemas/v1"
+	"github.com/grafana/phlare/pkg/util/build"
+)
+
+type profileGetter struct {
+	// first preference a direct profile reference
+	p *schemav1.Profile
+
+	// second try read from row group
+	rowGroup *rowGroupOnDisk
+	rowNum   int64
+}
+
+func (pg *profileGetter) Get() *schemav1.Profile {
+	if pg.p != nil {
+		return pg.p
+	}
+
+	if pg.rowGroup == nil {
+		return nil
+	}
+
+	reader := parquet.NewGenericRowGroupReader[*schemav1.Profile](pg.rowGroup)
+	if err := reader.SeekToRow(pg.rowNum - 1); err != nil {
+		// TODO: Log error here
+		panic(fmt.Errorf("unable to seek row_num=%d from %s: %w", pg.rowNum, pg.rowGroup.file.Name(), err))
+	}
+
+	profiles := make([]*schemav1.Profile, 1)
```

Comment:
At scale, reading one row at a time will be problematic I think we should use flushed files to search for profiles instead.

I have some doubts about using profileGetter. Usually, you are querying for many profiles at once.

Reply:
I guess then it probably would be best to keep the row numbers managed in the index, which contain a particular series. (rather than querying by row group).

Reply:
Yes definitively.